index.js

```diff
@@ -440,19 +486,43 @@ function collectObj (stream, cb) {
   function end () { cb(null, rows) }
 }
 
+// Object, (k, v -> v) -> Object
 function mapObj (obj, fn) {
   Object.keys(obj).forEach(function (key) {
     obj[key] = fn(key, obj[key])
   })
   return obj
 }
 
-function filterObj (obj, fn) {
-  var res = {}
-  Object.keys(obj).forEach(function (key) {
-    if (fn(key, obj[key])) {
-      res[key] = obj[key]
+// KdbPoint -> OsmVersion
+function kdbPointToVersion (pt) {
+  return pt.value.toString('hex')
+}
+
+// OsmDb, OsmVersion -> [OsmId]
+function getWayNodesOfDeletedDoc (osm, version, cb) {
```

Comment:
I've updated this logic. Sanity check to make sure we're on the same page re what happens now:

1. Do not recurse down to a deleted way's `refs` or relation's `members`.
2. If the node found in the kdb query is referred to only by deleted ways/relations, ignore the node and just return the deleted way/relation (and none of its `refs`/`members`).